api/src/opentrons/cli/analyze.py

```diff
@@ -78,12 +86,13 @@ async def _analyze(files_and_dirs: Sequence[Path], json_mode: bool) -> None:
             commands=analysis.commands,
             errors=analysis.errors,
         )
-        click.echo(results.json())
-        return
 
-    raise click.UsageError(
-        "Currently, this tool only supports JSON mode. Use `--json`."
-    )
+        await json_output.write_text(results.json(exclude_none=True))
```

Comment:
```suggestion
        await json_output.write_text(
            results.json(exclude_none=True),
            encoding="utf-8",
        )
```

I learned recently that if you omit the explicit `encoding` argument, the default is platform-dependent. 😬  We should probably explicitly specify `"utf-8"` everywhere, as a general practice.

(See [PEP 686](https://peps.python.org/pep-0686/) for a proposal to change this. Perhaps we'll be able to benefit from it by the time we're all living on Mars.)